pyro/contrib/gp/kernels/kernel.py

```diff
@@ -316,10 +316,11 @@ class Warping(Transforming):
     learning architecture. For example:
 
         >>> linear = torch.nn.Linear(10, 3)
-        # register its parameters to Pyro's ParamStore
-        >>> pyro_linear = pyro.module("linear", linear)
+        # register its parameters to Pyro's ParamStore and wrap it by lambda
+        # to call the primitive pyro.module each time we use the linear function
+        >>> pyro_linear_fn = lambda x: pyro.module("linear", linear)(x)
```

Comment:
Interesting. I guess this is only needed if `pyro_linear_fn` is defined outside of `model`?

Reply:
Yes. It is required when arguments are functions and those functions are `nn.Module`s. In case there is no parameter in that function or that function is a Parameterized class, we don't have to use lambda wrapper. :)